lib/transition/unpack.c

```diff
@@ -228,10 +498,31 @@ unpack_graph(xmlNode * xml_graph, const char *reference)
 
             if (new_synapse != NULL) {
                 new_graph->synapses = g_list_append(new_graph->synapses, new_synapse);
+                probe_actions = prepend_probe_action(probe_actions, new_synapse);
+                start_actions = prepend_start_action(start_actions, new_synapse);
```

Comment:
I might be missing something here, but wouldn't it be easier to just check all the actions here, and inject the metadata if !has_injected_metadata_for(action, injected) && task is resource action && (task is probe or task is start or !has_cached_metadata_for(action)? That would avoid making lists and looping through actions multiple times.